backend/lambdas/api/repo/repo/gitlab_util/process_gitlab_utils.py

```diff
@@ -2,41 +2,39 @@
 import json
 import re
 import urllib
-from string import Template
-
 import requests
+from graphql_query import Argument, Field, Operation, Query, Variable
 
 from repo.util.aws import AWSConnect
-from repo.util.const import GITLAB_QUERY_NO_BRANCH, GITLAB_QUERY_WITH_BRANCH
 from repo.util.env import DEFAULT_ORG, REV_PROXY_DOMAIN_SUBSTRING, REV_PROXY_SECRET_HEADER
 from repo.util.utils import GetProxySecret, Logger, auth
 
 log = Logger(__name__)
 
 
-def _process_query_list(key, service_url, query_list, batch_query=True):
+def process_query_list(key, service_url, query_list, vars, batch_query=True):
     if batch_query:
-        query = "{%s}" % " ".join(query_list)
-        return _get_query_response(key, service_url, query)
+        return _get_query_response(key, service_url, query_list, vars)
     response_dict = {"data": {}}
     for query_item in query_list:
-        query = "query {}".format(query_item)
-        resp = _get_query_response(key, service_url, query)
-        if resp and "data" in resp and "project" in resp.get("data"):
-            resp_data = resp["data"]["project"]
-            repo = re.match("repo[0-9]*", query_item.strip()).group(0)
-            response_dict["data"][repo] = resp_data
+        resp = _get_query_response(key, service_url, query_item, vars)
+        if resp and "data" in resp:
+            resp_data = resp["data"]
+            # Parsing out the alias from the GraphQL query, so we can use it for mapping.
+            # ex: query GetRepos($repo0: ID!) {repo0: project(fullPath: $repo0) {**}
+            repo = re.search("\$(repo[0-9]*):", query_item.strip()).group(0)
```

Comment:
`group(0)` returns the entire match.  I think this should be `group(1)` instead to return just the first capture group?

Reply:
Learned all sorts of stuff about python regex today